app/src/main/java/com/github/sdpcoachme/database/FireDatabase.kt

```diff
@@ -59,33 +59,34 @@ class FireDatabase(databaseReference: DatabaseReference) : Database {
 
         if (groupEvent.participants.size > groupEvent.maxParticipants) {
             errorPreventionFuture.completeExceptionally(Exception("Group event should not be full, initially"))
-        } else if (groupEvent.participants.isEmpty()) {
-            errorPreventionFuture.completeExceptionally(Exception("Group event must have at least 1 participants"))
         } else if (LocalDateTime.parse(groupEvent.event.start).isBefore(LocalDateTime.now())) {
             errorPreventionFuture.completeExceptionally(Exception("Group event cannot be in the past"))
         } else {
-            errorPreventionFuture = setChild(groupEvents, groupEvent.groupEventId, groupEvent).thenCompose {
-                registerForGroupEvent(groupEvent.organiser, groupEvent.groupEventId)
-            }
+            errorPreventionFuture = setChild(groupEvents, groupEvent.groupEventId, groupEvent)
```

Comment:
Great job on moving that piece of code to the CachingStore. Follows great logic: FireDatabase should provide easy database access methods without unpredictable side effects, therefore it is better to chain those actions in CachingStore.